mpp/shmem_c_func.m4

```diff
@@ -156,6 +156,32 @@ define(`SHMEM_C_SET',
 `SHMEM_FUNCTION_ATTRIBUTES void SHPRE()shmem_$1_set($2 *target, $2 value, int pe)')dnl
 SHMEM_DECLARE_FOR_EXTENDED_AMO(`SHMEM_C_SET')
 
+/* AMO: Atomic Bitwise Routines */
+define(`SHMEM_C_XOR',
+`SHMEM_FUNCTION_ATTRIBUTES void SHPRE()shmem_$1_atomic_xor($2 *target, $2 value, int pe)')dnl
+SHMEM_DECLARE_FOR_BITWISE_AMO(`SHMEM_C_XOR')
+
+define(`SHMEM_C_AND',
+`SHMEM_FUNCTION_ATTRIBUTES void SHPRE()shmem_$1_atomic_and($2 *target, $2 value, int pe)')dnl
+SHMEM_DECLARE_FOR_BITWISE_AMO(`SHMEM_C_AND')
+
+define(`SHMEM_C_OR',
+`SHMEM_FUNCTION_ATTRIBUTES void SHPRE()shmem_$1_atomic_or($2 *target, $2 value, int pe)')dnl
+SHMEM_DECLARE_FOR_BITWISE_AMO(`SHMEM_C_OR')
+
+/* AMO: Fetching atomic Bitwise Routines */
+define(`SHMEM_C_FETCH_XOR',
+`SHMEM_FUNCTION_ATTRIBUTES $2 SHPRE()shmem_$1_atomic_fetch_xor($2 *target, $2 value, int pe)')dnl
+SHMEM_DECLARE_FOR_BITWISE_AMO(`SHMEM_C_FETCH_XOR')
+
+define(`SHMEM_C_FETCH_AND',
+`SHMEM_FUNCTION_ATTRIBUTES $2 SHPRE()shmem_$1_atomic_fetch_and($2 *target, $2 value, int pe)')dnl
+SHMEM_DECLARE_FOR_BITWISE_AMO(`SHMEM_C_FETCH_AND')
+
+define(`SHMEM_C_FETCH_OR',
+`SHMEM_FUNCTION_ATTRIBUTES $2 SHPRE()shmem_$1_atomic_fetch_or($2 *target, $2 value, int pe)')dnl
+SHMEM_DECLARE_FOR_BITWISE_AMO(`SHMEM_C_FETCH_OR')
+
```

Comment:
I don't see any evidence of fetch_add and fetch_inc.  Are those accounted for in `SHMEM_C_FETCH`?   Elsewhere?

Reply:
Maybe all of the atomic_* routines are not address in this commit...  Just the bitwise?

Reply:
This is just the new types.  I pulled bitwise AMOs in since they were already done and I needed them to test part of the bindings generators.